### PR TITLE
Turtle brain and unit test, fix PyCQA/astroid#849

### LINKED ISSUE #849: astroid brain for turtle module
PyLint generates a lot of errors in a program using the turtle module: PyCQA/pylint#1928

The following brain fixes this. It uses the same method as the turtle module to generate stubs for astroid to parse.

**brain_turtle.py**
```
import astroid

def register(linter):
  pass

def transform():
  import turtle
  def _make_global_funcs(functions, cls):
    funcs = []
    for methodname in functions:
      method = getattr(cls, methodname)
      paramslist, argslist = turtle.getmethparlist(method)
      if paramslist == "": continue
      funcs.append(f"def {methodname}{paramslist}: return")
    return funcs
  funcs = []
  funcs.extend(_make_global_funcs(turtle._tg_screen_functions, turtle._Screen))
  funcs.extend(_make_global_funcs(turtle._tg_turtle_functions, turtle.Turtle))
  return astroid.parse('\n'.join(funcs))

astroid.register_module_extender(astroid.MANAGER, "turtle", transform)
```

### PR DESCRIPTION
**Pull request by rioj7**
## Steps

- [ ] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [x ] Write a good description on what the PR does.

## Description

Add a brain for the python Turtle module, and a unit test for this brain

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :sparkles: New feature |

### CODE REVIEW

File: astroid/brain/brain_turtle.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    method = getattr(cls, methodname)
+    paramslist, argslist = turtle.getmethparlist(method)
+    if paramslist == "": continue
+    funcs.append(f"def {methodname}{paramslist}: return")
+  return funcs
+funcs = []
+funcs.extend(_make_global_funcs(turtle._tg_screen_functions, turtle._Screen))
+funcs.extend(_make_global_funcs(turtle._tg_turtle_functions, turtle.Turtle))
+print('\n'.join(funcs))
+"""
```
Comment (hippo91) [inline astroid/brain/brain_turtle.py:19]: It is interesting but i'm not sure it is the right place to store this script. Why not in a proper repo on your github account, with a link toward it?

Comment (hippo91) [reply astroid/brain/brain_turtle.py:19]: In fact i'v got a similar script that i used for `numpy` but it doesn't make part of `astroid` sources.

File: tests/unittest_brain_turtle.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+class TurtleTest(unittest.TestCase):
+
+    def test_pencolor(self):
+        node = dedent("""
+            import turtle
+            turtle.pencolor
+            """)
+        inferred = next(astroid.extract_node(node).infer())
+        self.assertIsInstance(inferred, astroid.FunctionDef)
+        self.assertEqual(inferred.name, 'pencolor')
```
Comment (hippo91) [inline tests/unittest_brain_turtle.py:17]: Why just focusing on `pencolor` function?
It should be possible to loop over all the function you added in the brain and make for each of them the same tests you are doing here.

File: tests/unittest_brain_turtle.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    def test_Turtle(self):
+        node = dedent("""
+            import turtle
+            turtle.Turtle()
+            """)
+        inferred = next(astroid.extract_node(node).infer())
+        self.assertIsInstance(inferred, astroid.Instance)
```
Comment (hippo91) [inline tests/unittest_brain_turtle.py:25]: If i understand well, you are not testing the brain you added but the classical `astroid` behavior (i.e not using the brain).
Am i correct?
If yes, i don't see the point.

**Review by hippo91 — CHANGES_REQUESTED** [review]
@rioj7 thanks a lot for this PR.
It's almost goog, i just left a pair of comments.
By the way, the CI is failing because `black` thinks there is something to reformat.

**Comment by Pierre-Sassoulas** [comment]
Closing as stale / abandoned. Feel free to open another MR if the issue still need addressing.

**Pull request closed without merging**